apis/v1alpha2/httproute_types.go

```diff
@@ -214,18 +214,33 @@ type HTTPRouteRule struct {
 //
 // Prefix and Exact paths must be syntactically valid:
 //
-// - Must begin with the '/' character
-// - Must not contain consecutive '/' characters (e.g. /foo///, //).
-// - For prefix paths, a trailing '/' character in the Path is ignored,
-// e.g. /abc and /abc/ specify the same match.
+// - Must begin with the `/` character
+// - Must not contain consecutive `/` characters (e.g. `/foo///`, `//`).
 //
 // +kubebuilder:validation:Enum=Exact;Prefix;RegularExpression
 type PathMatchType string
 
-// PathMatchType constants.
 const (
-	PathMatchExact             PathMatchType = "Exact"
-	PathMatchPrefix            PathMatchType = "Prefix"
+	// Matches based on a URL path prefix split by `/`. Matching is
+	// case sensitive and done on a path element by element basis. A
+	// path element refers to the list of labels in the path split by
+	// the `/` separator. A request is a match for path _p_ if every
+	// _p_ is an element-wise prefix of the request path.
+	//
+	// For example, `/abc`, `/abc/` and `/abc/def` match the prefix
+	// `/abc`, but `/abcd` does not.
+	PathMatchExact PathMatchType = "Exact"
+
+	// Matches the URL path exactly and with case sensitivity.
+	PathMatchPrefix PathMatchType = "Prefix"
```

Comment:
@jpeach :wave: was just looking at this and the godoc for these two (`Exact` and `Prefix`) are switched.

Reply:
![](https://media.giphy.com/media/UsN7AnMBFjLSiTSaYN/giphy.gif)